src/config/syscheck-config.c

```diff
@@ -947,9 +947,14 @@ static int read_attr(syscheck_config *syscheck, const char *dirs, char **g_attrs
                 strchr(real_path, '?') ||
                 strchr(real_path, '[')) {
             int gindex = 0;
+            int gstatus;
             glob_t g;
 
-            if (glob(tmp_dir, 0, NULL, &g) != 0) {
+            if (gstatus = glob(tmp_dir, 0, NULL, &g), gstatus == GLOB_NOMATCH) {
```

Comment:
Variable set + comparison is correct, but this syntax makes Cppcheck produce a false positive.

```suggestion
            if (glob(tmp_dir, 0, NULL, &g) == GLOB_NOMATCH) {
```